mne/time_frequency/_stockwell.py

```diff
@@ -171,36 +163,43 @@ def _induced_power_stockwell(data, sfreq, fmin, fmax, n_fft=None, width=1.0,
 
     start_f = np.abs(freqs - fmin).argmin()
     stop_f = np.abs(freqs - fmax).argmin()
+    freqs = freqs[start_f:stop_f]
 
-    M, f_range, windows = _precompute_st_windows(data.shape[-1], start_f,
-                                                 stop_f, sfreq, width)
-    n_frequencies = len(f_range)
-    psd = np.empty((n_channels, n_frequencies, n_times))
-    itc = np.empty((n_channels, n_frequencies, n_times))
-
-    n_jobs = check_n_jobs(n_jobs)
-    parallel, my_st, _ = parallel_func(_st_power_itc, n_jobs)
-    tfrs = parallel(my_st(data[:, c, :], M, f_range, windows,
-                          return_itc, zero_pad, decim, n_times)
-                    for c in range(n_channels))
-    tfrs = iter(tfrs)
-    for i_chan, (this_psd, this_itc) in enumerate(tfrs):
-        psd[i_chan] = this_psd
-        if this_itc is not None:
-            itc[i_chan] = this_itc
+    W = _precompute_st_windows(data.shape[-1], start_f, stop_f, sfreq, width)
+    n_freq = stop_f - start_f
+    psd = np.empty((n_channels, n_freq, n_out))
+    itc = np.empty((n_channels, n_freq, n_out)) if return_itc else None
+
+    if n_jobs == 1:
+        for c in range(n_channels):
+            psd[c], this_itc = \
+                _st_power_itc(data[:, c, :], start_f, return_itc, zero_pad,
+                              decim, W)
+            if return_itc:
+                itc[c] = this_itc
+    else:
+        parallel, my_st, _ = parallel_func(_st_power_itc, n_jobs)
+        tfrs = parallel(my_st(data[:, c, :], start_f, return_itc, zero_pad,
+                              decim, W)
+                        for c in range(n_channels))
+        for c, (this_psd, this_itc) in enumerate(iter(tfrs)):
+            psd[c] = this_psd
+            if this_itc is not None:
+                itc[c] = this_itc
```

Comment:
I think it will cause memory usage to go up, just like in the other cases where you suggested it needed to branch. Are you sure combining them won't increase memory usage? In any case, I made this consistent with the other two methods, so we should either branch all of them, or none of them...